File: reverse-linked-list.cc
// Reverse Linked List
class Solution {
public:
  ListNode* reverseList(ListNode* x) {
    ListNode *y = 0, *t;
    while (x) {
      t = x->next;
      x->next = y;
      y = x;
      x = t;
    }
    return y;
  }
};

File: reverse-linked-list.nim
type ListNode {.importc.} = ptr object
  val: cint
  next: ListNode

proc reverseList(head: ListNode): ListNode {.exportc.} =
  var x = head
  while x != nil:
    let y = x.next
    x.next = result
    result = x
    x = y
